packages/core/src/logger/apiLogger.ts

```diff
@@ -5,45 +5,175 @@ import {
+    const contentLengthHeader = this.getContentLength(response.headers);
+
     this._logger.log(
-      Level.Info,
-      'Response ',
-      { HttpStatusCode: coreResponse.statusCode },
-      { bodyLength: coreResponse.body.toString().length }
+      logLevel,
+      `Response HttpStatusCode ${response.statusCode} Length ${contentLengthHeader} ContentType ${contentTypeHeader}`,
+      {
+        statusCode: response.statusCode,
+        contentLength: contentLengthHeader,
+        contentType: contentTypeHeader,
+      }
     );
+
+    if (this._loggingOptions.logResponse) {
+      const {
+        logBody,
+        logHeaders,
+        headerToInclude,
+        headerToExclude,
+      } = this._loggingOptions.logResponse;
+
+      if (logHeaders) {
+        const headersToLog = this.extractHeadersToLog(
+          response.headers,
+          headerToInclude,
+          headerToExclude
+        );
+
+        this._logger.log(logLevel, `Response Headers ${headersToLog}`, {
+          headers: headersToLog,
+        });
+      }
+
+      if (logBody) {
+        this._logger.log(logLevel, `Response Body ${response.body}`, {
+          body: response.body,
+        });
+      }
+    }
   }
-}
 
-export const requestLoggerInterceptor = (apiLogger: ApiLoggerInterface) => {
-  return (request, options, next) => {
-    apiLogger.logRequest(request);
-    return next(request, options);
-  };
-};
-
-export const responseLoggerInterceptor = (apiLogger: ApiLoggerInterface) => {
-  return (context) => {
-    apiLogger.logResponse(context);
-    return context;
-  };
-};
+  private getContentType(headers?: Record<string, string>): string {
+    return headers ? getHeader(headers, CONTENT_TYPE_HEADER) ?? '' : '';
+  }
+
+  private getContentLength(headers?: Record<string, string>): string {
+    return headers ? getHeader(headers, CONTENT_LENGTH_HEADER) ?? '' : '';
+  }
+
+  private removeQueryParams(url: string): string {
+    const queryStringIndex: number = url.indexOf('?');
+    return queryStringIndex !== -1 ? url.substring(0, queryStringIndex) : url;
+  }
+
+  private extractHeadersToLog(
+    headers?: Record<string, string>,
+    headersToInclude?: string[],
+    headersToExclude?: string[]
+  ): Record<string, string> {
+    if (!headers) {
+      return {};
+    }
+
+    if (headersToInclude && headersToInclude.length > 0) {
+      // Filter headers based on the keys specified in headersToInclude
+      const filteredHeaders = Object.entries(headers)
+        .filter(([key]) => headersToInclude.includes(key))
+        .reduce((acc, [key, value]) => {
+          acc[key] = value;
+          return acc;
+        }, {});
+
+      return filteredHeaders;
+    }
+
+    if (headersToExclude && headersToExclude.length > 0) {
+      // Filter headers based on the keys specified in headersToExclude
+      const filteredHeaders = Object.entries(headers)
+        .filter(([key]) => !headersToExclude.includes(key))
```

Comment:
Header names should be matched via case-insensitive string comparison.

Also, please check if our httpHeaders utilities contain something that we can use here. It seems like we are reimplementing header lookup logic.